types/contracts/Poap.client.ts

```diff
@@ -0,0 +1,213 @@
+  client: CosmWasmClient;
+  contractAddress: string;
+
+  constructor(client: CosmWasmClient, contractAddress: string) {
+    this.client = client;
+    this.contractAddress = contractAddress;
+    this.config = this.config.bind(this);
+    this.eventInfo = this.eventInfo.bind(this);
+    this.mintedAmount = this.mintedAmount.bind(this);
+    this.allNftInfo = this.allNftInfo.bind(this);
+    this.tokens = this.tokens.bind(this);
+  }
+
+  config = async (): Promise<QueryConfigResponse> => {
+    return this.client.queryContractSmart(this.contractAddress, {
+      config: {}
+    });
+  };
+  eventInfo = async (): Promise<QueryEventInfoResponse> => {
+    return this.client.queryContractSmart(this.contractAddress, {
+      event_info: {}
+    });
+  };
+  mintedAmount = async ({
+    user
+  }: {
+    user: string;
+  }): Promise<QueryMintedAmountResponse> => {
+    return this.client.queryContractSmart(this.contractAddress, {
+      minted_amount: {
+        user
+      }
+    });
+  };
+  allNftInfo = async ({
+    includeExpired,
+    tokenId
+  }: {
+    includeExpired?: boolean;
+    tokenId: string;
+  }): Promise<AllNftInfoResponseForMetadata> => {
+    return this.client.queryContractSmart(this.contractAddress, {
+      all_nft_info: {
+        include_expired: includeExpired,
+        token_id: tokenId
+      }
+    });
+  };
+  tokens = async ({
+    limit,
+    owner,
+    startAfter
+  }: {
+    limit?: number;
+    owner: string;
+    startAfter?: string;
+  }): Promise<TokensResponse> => {
+    return this.client.queryContractSmart(this.contractAddress, {
+      tokens: {
+        limit,
+        owner,
+        start_after: startAfter
+      }
+    });
+  };
+}
+export interface PoapInterface extends PoapReadOnlyInterface {
+  contractAddress: string;
+  sender: string;
+  enableMint: (fee?: number | StdFee | "auto", memo?: string, funds?: Coin[]) => Promise<ExecuteResult>;
+  disableMint: (fee?: number | StdFee | "auto", memo?: string, funds?: Coin[]) => Promise<ExecuteResult>;
+  mint: (fee?: number | StdFee | "auto", memo?: string, funds?: Coin[]) => Promise<ExecuteResult>;
+  mintTo: ({
+    recipient
+  }: {
+    recipient: string;
+  }, fee?: number | StdFee | "auto", memo?: string, funds?: Coin[]) => Promise<ExecuteResult>;
+  updateEventInfo: ({
+    endTime,
+    startTime
+  }: {
+    endTime: Timestamp;
+    startTime: Timestamp;
+  }, fee?: number | StdFee | "auto", memo?: string, funds?: Coin[]) => Promise<ExecuteResult>;
+  updateAdmin: ({
+    newAdmin
+  }: {
+    newAdmin: string;
+  }, fee?: number | StdFee | "auto", memo?: string, funds?: Coin[]) => Promise<ExecuteResult>;
+  updateMinter: ({
+    newMinter
+  }: {
+    newMinter: string;
+  }, fee?: number | StdFee | "auto", memo?: string, funds?: Coin[]) => Promise<ExecuteResult>;
+}
+export class PoapClient extends PoapQueryClient implements PoapInterface {
+  override client: SigningCosmWasmClient;
```

Comment:
The override now is added manually since the generate tool has the issue:
https://github.com/CosmWasm/ts-codegen/issues/78